### PR TITLE
feat: Add Go code generation option to Eventbridge Schemas workflow

### PR DESCRIPTION
**Pull request by kolaamoo25**
#### Which issue(s) does this change fix?
No issue number attached for this PR

#### Why is this change necessary?
Allows users to create EventBridge Schema applications using the go1.x runtime

#### How does it address the issue?
This change maps the go1.x runtime to the Go enum value needed to call the Schemas Download CodeBinding API

#### What side effects does this change have?
N/A

#### Checklist

- [] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [] Write unit tests
- [] Write/update functional tests
- [X] Write/update integration tests
- [X] `make pr` passes
- [] `make update-reproducible-reqs` if dependencies were changed
- [] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: tests/integration/init/schemas/test_init_with_schemas_command.py
Comment (moelasmar) [inline tests/integration/init/schemas/test_init_with_schemas_command.py:223]: please add another use case to test the `event bridge schema app` template, and mark it `@skipif(SKIP_SCHEMA_TESTS)`

**Review by moelasmar — CHANGES_REQUESTED** [review]

File: tests/integration/init/schemas/test_init_with_schemas_command.py
Comment (moelasmar) [reply tests/integration/init/schemas/test_init_with_schemas_command.py:223]: sorry I got confused, so tag this one with `@skipif(SKIP_SCHEMA_TESTS)`

Comment (kolaamoo25) [reply tests/integration/init/schemas/test_init_with_schemas_command.py:223]: This entire test class is already being skipped by this line: https://github.com/aws/aws-sam-cli/blob/develop/tests/integration/init/schemas/test_init_with_schemas_command.py#L17

File: samcli/local/common/runtime_template.py
Comment (kolaamoo25) [inline samcli/local/common/runtime_template.py:167]: Dotnet is currently not a supported language for EventBridge Schemas code bindings

**Review by jfuss — APPROVED** [review]

**Comment by jfuss** [comment]
@kolaamoo I know we have app templates pr opened as well but I don't understand the order these need to be released (and merged). Do we need this merged and released first before merging app templates?

**Comment by kolaamoo25** [comment]
> @kolaamoo I know we have app templates pr opened as well but I don't understand the order these need to be released (and merged). Do we need this merged and released first before merging app templates?

If my understanding it correct, the way the sam-cli works is that it clones the sam-cli-app-templates repo for every sam init call. So the sam-cli-app-templates deployment time is pretty much instant after it's approved and merged. There's nothing blocking the merge and deployment on the sam-cli side before sam-cli-app-templates. Let me know if I have something wrong.

File: samcli/local/common/runtime_template.py
Comment (hawflau) [reply samcli/local/common/runtime_template.py:145]: minor - If it's not supported, shall we remove this line? Even if it will be supported in the future, I bet the language code would be "dotnetcore3.1". It doesn't hurt to still keep it here as it doesn't impact anything, but just create confusion.

Comment (kolaamoo25) [reply samcli/local/common/runtime_template.py:145]: I think some other tests rely on this value being here. I tried removing the value and some of the checks failed.

**Review by torresxb1 — APPROVED** [review]

**Review by moelasmar — APPROVED** [review]

**Comment by torresxb1** [comment]
Thanks for your contribution! @kolaamoo25

**Pull request merged**
